src/app/core/store/collections/collections.effects.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Injectable} from '@angular/core';
+import {Actions, Effect} from '@ngrx/effects';
+import {Action} from '@ngrx/store';
+import {Observable} from 'rxjs/Observable';
+import {catchError, map, switchMap, tap} from 'rxjs/operators';
+import {Collection} from '../../dto/collection';
+import {Permission} from '../../dto/permission';
+import {CollectionService} from '../../rest/collection.service';
+import {SearchService} from '../../rest/search.service';
+import {QueryConverter} from '../navigation/query.converter';
+import {NotificationsAction} from '../notifications/notifications.action';
+import {PermissionsConverter} from '../permissions/permissions.converter';
+import {PermissionModel, PermissionType} from '../permissions/permissions.model';
+import {CollectionConverter} from './collection.converter';
+import {AttributeModel, CollectionModel} from './collection.model';
+import {CollectionsAction, CollectionsActionType} from './collections.action';
+
+@Injectable()
+export class CollectionsEffects {
+
+  @Effect()
+  public get$: Observable<Action> = this.actions$.ofType<CollectionsAction.Get>(CollectionsActionType.GET).pipe(
+    switchMap((action) => {
+      const queryDto = QueryConverter.toDto(action.payload.query);
+
+      return this.searchService.searchCollections(queryDto).pipe(
+        map((dtos: Collection[]) => dtos.map(dto => CollectionConverter.fromDto(dto)))
+      );
+    }),
+    map((collections: CollectionModel[]) => new CollectionsAction.GetSuccess({collections: collections})),
+    catchError((error) => Observable.of(new CollectionsAction.GetFailure({error: error})))
+  );
+
+  @Effect()
+  public getFailure$: Observable<Action> = this.actions$.ofType<CollectionsAction.GetFailure>(CollectionsActionType.GET_FAILURE).pipe(
+    tap(action => console.error(action.payload.error)),
```

Comment:
Do we want to write errors to console?

Reply:
In the future, we would need some centralized solution which will send these errors to our servers. In the meantime, I think logging is the only solution, or do you have any suggestion how to handle them?

Reply:
No, it's OK.